lib/ansible/modules/cloud/amazon/aws_kms.py

```diff
@@ -625,134 +607,173 @@ def compare_grants(existing_grants, desired_grants, purge_grants=False):
+        module.fail_json_aws(e, msg="Failed to cancel key deletion")
 
+    return True
 
-def update_alias(connection, module, key_id, alias):
-    if not alias.startswith('alias/'):
-        alias = 'alias/' + alias
-    aliases = get_kms_aliases_with_backoff(connection)['Aliases']
-    if key_id:
-        # We will only add new aliases, not rename existing ones
-        if alias not in [_alias['AliasName'] for _alias in aliases]:
+
+def ensure_enabled_disabled(connection, module, key, enabled):
+    desired_state = ('Disabled', 'Enabled')[enabled]
+    if key['key_state'] == desired_state:
+        return False
+
+    key_id = key['key_arn']
+    if not module.check_mode:
+        if enabled:
             try:
-                connection.create_alias(KeyId=key_id, AliasName=alias)
-                return True
+                connection.enable_key(KeyId=key_id)
             except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
-                module.fail_json_aws(e, msg="Failed create key alias")
-    return False
+                module.fail_json_aws(e, msg="Failed to enable key")
+        else:
+            try:
+                connection.disable_key(KeyId=key_id)
+            except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+                module.fail_json_aws(e, msg="Failed to disable key")
 
+    return True
 
-def update_key(connection, module, key):
-    changed = False
-    alias = module.params.get('alias')
-    key_id = key['key_arn']
 
-    if alias:
-        changed = update_alias(connection, module, key_id, alias) or changed
+def update_alias(connection, module, key, alias):
+    alias = canonicalize_alias_name(alias)
+
+    if alias is None:
+        return False
+
+    key_id = key['key_arn']
+    aliases = get_kms_aliases_with_backoff(connection)['Aliases']
+    # We will only add new aliases, not rename existing ones
+    if alias in [_alias['AliasName'] for _alias in aliases]:
+        return False
 
-    if key['key_state'] == 'PendingDeletion':
+    if not module.check_mode:
         try:
-            connection.cancel_key_deletion(KeyId=key_id)
-            # key is disabled after deletion cancellation
-            # set this so that ensure_enabled_disabled works correctly
-            key['key_state'] = 'Disabled'
-            changed = True
+            connection.create_alias(TargetKeyId=key_id, AliasName=alias)
         except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
-            module.fail_json_aws(e, msg="Failed to cancel key deletion")
+            module.fail_json_aws(e, msg="Failed create key alias")
+
+    return True
 
-    changed = ensure_enabled_disabled(connection, module, key) or changed
 
-    description = module.params.get('description')
-    # don't update description if description is not set
-    # (means you can't remove a description completely)
```

Comment:
Not changing something when it's not set is the standard behavior, so I didn't think it was worth documenting.  Additionally AWS only allows you to set it to an empty string rather than totally remove it, passing an empty string (rather than not passing anything) will set it to an empty string.